wakuv2/waku.go

```diff
@@ -1650,16 +1678,54 @@ func (w *Waku) seedBootnodesForDiscV5() {
 		return lastTry+backoff < now()
 	}
 
+	// Ensure that only a single dnsDiscAsyncRetrievedSignal is processed at the time
+	// to avoid restarting in fast succession
+	dnsDiscWorkSignalReceived := make(chan struct{})
+	go func() {
+		t := time.NewTicker(3 * time.Second)
+		defer t.Stop()
+
+		dnsDiscAsyncRetrieved := false
+		for {
+			select {
+			case <-w.ctx.Done():
+				return
+			case <-w.dnsDiscAsyncRetrievedSignal:
+				dnsDiscAsyncRetrieved = true
+				t.Reset(3 * time.Second)
+			case <-t.C:
+				if dnsDiscAsyncRetrieved {
```

Comment:
Wondering why not execute this block of code immediately when get the signal `dnsDiscAsyncRetrievedSignal`? Even call `restartDiscV5` directly, the channels here feels complex.
If there is some race condition, we can probably add a lock.

Reply:
I used this logic because i thought that executing discv5 in fast succession is not ideal (this could happen only assuming that you have more than one enrtree registered that failed to be resolved). But I agree that it adds complexity. I'll remove the need for a ticker/select